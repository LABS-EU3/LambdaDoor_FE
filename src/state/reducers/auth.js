import * as types from '../types';

const initialState = {
  isLoggingIn: false,
  isLoggedIn: false,
};

export const authState = (state = initialState, action) => {
  switch (action.type) {
    case types.LOG_IN_USER:
      return {
        ...state,
        isLoggingIn: true,
      };
    case types.GET_COMPANY_REVIEWS_SUCCESS:
      return {
        ...state,
        isLoggedIn: true,
      };
    case types.GET_COMPANY_REVIEWS_FAILURE:
      return state;
    default:
      return state;
  }
};
