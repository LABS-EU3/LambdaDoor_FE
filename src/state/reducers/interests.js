import * as types from '../types';
import { allInterests } from '../../utils/data';

const initialState = {
  interests: [],
};

// eslint-disable-next-line import/prefer-default-export
export const interestReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_INTERESTS_SUCCESS:
      return {
        ...state,
        isLoading: false,
        interests: action.payload,
      };

    case types.DELETE_INTEREST:
      console.log(action.payload);
      return {
        ...state,
        interests: state.interests.filter(
          interest => interest.id !== action.payload
        ),
      };

    case types.ADD_INTEREST:
      return {
        ...state,
        interests: [
          ...state.interests,
          { id: action.payload, interest: allInterests[action.payload] },
        ],
      };
    default:
      return state;
  }
};
