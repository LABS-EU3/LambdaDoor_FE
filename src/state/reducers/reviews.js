import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: [],
};

export const companyReviews = (state = {}, action) => {
  switch (action.type) {
    case types.GET_COMPANY_REVIEWS:
      return {};
    case types.GET_COMPANY_REVIEWS_SUCCESS:
      return {
        ...state,
        message: action.payload.message,
      };
    case types.GET_COMPANY_REVIEWS_FAILURE:
      return {};
    default:
      return state;
  }
};

export const salaryReviews = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SALARY_REVIEWS:
      return { ...state, isFetching: true };
    case types.GET_SALARY_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: [...state.reviews, action.payload],
        isFetching: false,
      };
    case types.GET_SALARY_REVIEWS_FAILURE:
      return { ...state, isFetching: false };
    default:
      return state;
  }
};

export const interviewReviews = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_INTERVIEW_REVIEWS:
      return { ...state, isFetching: true };
    case types.GET_INTERVIEW_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: [...state.reviews, action.payload],
      };
    case types.GET_INTERVIEW_REVIEWS_FAILURE:
      return { ...state, isFetching: false };
    default:
      return state;
  }
};

export const reviewsError = (state = null, action) => {
  switch (action.type) {
    case types.GET_COMPANY_REVIEWS_FAILURE:
    case types.GET_SALARY_REVIEWS_FAILURE:
    case types.GET_INTERVIEW_REVIEWS_FAILURE:
      return action.payload;
    case types.GET_COMPANY_REVIEWS:
    case types.GET_SALARY_REVIEWS:
    case types.GET_INTERVIEW_REVIEWS:
      return null;
    case types.GET_COMPANY_REVIEWS_SUCCESS:
    case types.GET_SALARY_REVIEWS_SUCCESS:
    case types.GET_INTERVIEW_REVIEWS_SUCCESS:
      return null;
    default:
      return state;
  }
};
