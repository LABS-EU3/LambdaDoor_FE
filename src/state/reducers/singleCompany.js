import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: {
    companyReview: [],
    // salaryReview: [],
    // interviewProcess: [],
  },
};

const companyReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SINGLE_COMPANY_REVIEWS:
      return { ...state, isFetching: true };

    case types.GET_SINGLE_COMPANY_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: {
          ...state.reviews,
          companyReview: action.payload,
        },
      };

    case types.GET_COMPANY_REVIEWS_FAILURE:
      return { ...state, isFetching: false };

    default:
      return state;
  }
};

export default companyReviewsReducer;
