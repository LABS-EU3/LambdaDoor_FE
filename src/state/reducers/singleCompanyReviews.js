/* eslint-disable import/prefer-default-export */
import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: {
    company: [],
    salary: [],
    interview: [],
  },
};

export const singleCompanyReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SINGLE_COMPANY_INTERVIEW_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: {
          ...state.reviews,
          interview: action.payload,
        },
      };

    default:
      return state;
  }
};
