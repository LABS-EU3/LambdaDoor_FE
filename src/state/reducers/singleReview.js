import * as types from '../types';

const initialState = {
  isFetching: false,
  reviews: {
    companyReview: [],
  },
};

const singleReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SINGLE_REVIEWS:
      return { ...state, isFetching: true };

    case types.GET_SINGLE_REVIEWS_SUCCESS:
      return {
        ...state,
        reviews: {
          ...state.reviews,
          companyReview: action.payload,
        },
      };

    case types.GET_SINGLE_REVIEWS_FAILURE:
      return { ...state, isFetching: false };

    default:
      return state;
  }
};

export default singleReviewsReducer;
