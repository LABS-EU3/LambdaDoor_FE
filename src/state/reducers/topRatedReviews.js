import * as types from '../types';

const initialState = {
  isfetching: false,
  topRatedReviews: [],
};
// eslint-disable-next-line import/prefer-default-export
export const topRatedReviewsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_TOP_RATED:
      return { ...state, isFetching: true };
    case types.GET_TOP_RATED_SUCCESS:
      return {
        isfetching: false,
        topRatedReviews: action.payload,
      };
    case types.GET_TOP_RATED_FAILURE:
      return { ...state, isFetching: false };
    default:
      return state;
  }
};
