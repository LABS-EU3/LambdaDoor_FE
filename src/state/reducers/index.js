import { combineReducers } from 'redux';
import { reviewsReducer, reviewsError } from './reviews';
import { authState } from './auth';
import { interestReducer } from './interests';
import { companiesReducer } from './companies';
import { jobrolesReducer } from './jobroles';
import companyReviewsReducer from './singleCompany';

const rootReducer = combineReducers({
  reviews: reviewsReducer,
  reviewsError: reviewsError,
  authState: authState,
  interests: interestReducer,
  companies: companiesReducer,
  jobroles: jobrolesReducer,
  singleCompanyReviews: companyReviewsReducer,
});

export default rootReducer;
