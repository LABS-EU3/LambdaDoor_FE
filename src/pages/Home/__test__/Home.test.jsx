import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import reducers from '../../../state/reducers/index';
import { renderWithRedux } from '../../../utils/testingHelpers';
import ConnectedHome, { Home } from '../Home';

const state = reducers(undefined, {});
const token =
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6OSwiaWF0IjoxNTc4NTY4NTgxfQ.6cR-MJGGZRHgszj1o1IgvyXfhEla3NYNegRT7aafXps';

describe('Home', () => {
  it('render correctly', () => {
    expect(rtl.render(<Home />).baseElement).toMatchSnapshot();
  });

  it('shows the name of the site', () => {
    const siteTitle = rtl.render(<Home />).queryByText(/lambda door/i);
    expect(siteTitle).toBeInTheDocument();
  });

  it('shows the tagline of the site', () => {
    const siteTagline = rtl
      .render(<Home />)
      .queryByText(
        'The one-stop portal for Lambda graduates looking for company information in the quest for a job.'
      );
    expect(siteTagline).toBeInTheDocument();
  });
});

describe('ConnectedHome', () => {
  let useEffect;
  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };

  beforeEach(() => {});
  useEffect = jest.spyOn(React, 'useEffect');
  mockUseEffect();
  global.localStorage = jest.fn().mockImplementation(() => {
    return {
      getItem: jest.fn().mockReturnValue(token),
    };
  });

  it('renders correctly', () => {
    expect(renderWithRedux(<ConnectedHome />).baseElement).toMatchSnapshot();
  });
  it('has the correct state on load', () => {
    expect(state).toEqual({
      reviews: {
        isFetching: false,
        reviews: { company: [], salary: [], interview: [] },
      },
      reviewsError: null,
      authState: {
        isLoading: true,
        isLoggedIn: false,
        credentials: {},
        error: null,
      },
    });
  });
  // it('decodes the token of a previously authenticated user and passes them to the dashboard', async () => {
  //   if (token) {
  //     await store.dispatch(auth.setAuthenticated(mockLoginData));
  //     expect(state.authState.isLoggedIn).toEqual(true);
  //   }
  // });
});
