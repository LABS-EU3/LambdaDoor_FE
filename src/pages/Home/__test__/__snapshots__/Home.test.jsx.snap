// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedHome renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bwzfXH fANWRT"
    >
      <div
        class="sc-htpNat eNPuYj"
      >
        <div
          class="sc-bdVaJa hECGmh"
        >
          <img
            alt="Lambda School"
            src="lambda-logo.png"
          />
        </div>
        <div
          class="sc-bxivhb dMFhYj"
        >
          <h1
            class="ant-typography siteTitle"
          >
            Lambda Door
          </h1>
          <div
            class="ant-typography"
          >
            The one-stop portal for Lambda graduates looking for company information in the quest for a job.
          </div>
          <a
            href="https://slack.com/oauth/authorize?scope=identity.basic,identity.email,identity.team,identity.avatar&client_id=154966377728.878377460261&redirect_uri=http://localhost:3000/dashboard"
          >
            <img
              alt="Sign in with Slack"
              height="40"
              src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
              srcset="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
              width="172"
            />
          </a>
        </div>
        <div
          class="ant-typography"
          style="color: white;"
        >
          Built by Lambda students, for Lambda students.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Home Displays a snapshot for the home page with it's image information 1`] = `
<div>
  <div
    class="sc-bwzfXH fANWRT"
  >
    <div
      class="sc-htpNat eNPuYj"
    >
      <div
        class="sc-bdVaJa hECGmh"
      >
        <img
          alt="Lambda School"
          src="lambda-logo.png"
        />
      </div>
      <div
        class="sc-bxivhb dMFhYj"
      >
        <h1
          class="ant-typography siteTitle"
        >
          Lambda Door
        </h1>
        <div
          class="ant-typography"
        >
          The one-stop portal for Lambda graduates looking for company information in the quest for a job.
        </div>
        <a
          href="https://slack.com/oauth/authorize?scope=identity.basic,identity.email,identity.team,identity.avatar&client_id=154966377728.878377460261&redirect_uri=http://localhost:3000/dashboard"
        >
          <img
            alt="Sign in with Slack"
            height="40"
            src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
            srcset="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
            width="172"
          />
        </a>
      </div>
      <div
        class="ant-typography"
        style="color: white;"
      >
        Built by Lambda students, for Lambda students.
      </div>
    </div>
  </div>
</div>
`;

exports[`Home Displays a snapshot for the home page with it's image information 2`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH fANWRT"
  >
    <div
      class="sc-htpNat eNPuYj"
    >
      <div
        class="sc-bdVaJa hECGmh"
      >
        <img
          alt="Lambda School"
          src="lambda-logo.png"
        />
      </div>
      <div
        class="sc-bxivhb dMFhYj"
      >
        <h1
          class="ant-typography siteTitle"
        >
          Lambda Door
        </h1>
        <div
          class="ant-typography"
        >
          The one-stop portal for Lambda graduates looking for company information in the quest for a job.
        </div>
        <a
          href="https://slack.com/oauth/authorize?scope=identity.basic,identity.email,identity.team,identity.avatar&client_id=154966377728.878377460261&redirect_uri=http://localhost:3000/dashboard"
        >
          <img
            alt="Sign in with Slack"
            height="40"
            src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
            srcset="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
            width="172"
          />
        </a>
      </div>
      <div
        class="ant-typography"
        style="color: white;"
      >
        Built by Lambda students, for Lambda students.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bwzfXH fANWRT"
    >
      <div
        class="sc-htpNat eNPuYj"
      >
        <div
          class="sc-bdVaJa hECGmh"
        >
          <img
            alt="Lambda School"
            src="lambda-logo.png"
          />
        </div>
        <div
          class="sc-bxivhb dMFhYj"
        >
          <h1
            class="ant-typography siteTitle"
          >
            Lambda Door
          </h1>
          <div
            class="ant-typography"
          >
            The one-stop portal for Lambda graduates looking for company information in the quest for a job.
          </div>
          <a
            href="https://slack.com/oauth/authorize?scope=identity.basic,identity.email,identity.team,identity.avatar&client_id=154966377728.878377460261&redirect_uri=http://localhost:3000/dashboard"
          >
            <img
              alt="Sign in with Slack"
              height="40"
              src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
              srcset="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
              width="172"
            />
          </a>
        </div>
        <div
          class="ant-typography"
          style="color: white;"
        >
          Built by Lambda students, for Lambda students.
        </div>
      </div>
    </div>
  </div>
</body>
`;
